src/emulator/auth/operations.ts

```diff
@@ -1048,6 +1056,72 @@ export function setAccountInfoImpl(
   });
 }
 
+function sendOobForEmailReset(state: ProjectState, initialEmail: string, url: URL) {
+  const mode: string = "recoverEmail";
+  const { oobRecord, maybeMessage } = createOobRecord(state, initialEmail, url, {
+    requestType: "RECOVER_EMAIL",
+    mode,
+  });
+
+  // Print out a developer-friendly log
+  assert(maybeMessage, "Internal asertion error: No message to be logged for RECOVER_EMAIL");
+  EmulatorLogger.forEmulator(Emulators.AUTH).log("BULLET", maybeMessage!);
+}
+
+function createOobRecord(
+  state: ProjectState,
+  email: string,
+  url: URL,
+  params: {
+    requestType: OobRequestType;
+    mode: string;
+    continueUrl?: string;
+  }
+): {
+  oobRecord: OobRecord;
+  maybeMessage?: string;
```

Comment:
Let's split logging into another function, maybe `logOobRecord` or something that just logs an OOB as a side effect. It's much cleaner that way.

Reply:
Done.